src/shared/components/GeneSelectionBox/GeneSymbolValidator.tsx

```diff
@@ -154,8 +155,16 @@ export default class GeneSymbolValidator extends React.Component<
     @computed get oql() {
         return getOQL(this.props.geneQuery);
     }
+
     @computed private get geneIds(): string[] {
-        return this.oql.error ? [] : this.oql.query.map(line => line.gene);
+        return this.oql.error
+            ? []
+            : _(this.oql.query)
+                  .flatMap(singleGeneQuery =>
+                      getGenesFromSingleGeneQuery(singleGeneQuery)
```

Comment:
getGenesFromComplexGeneQuery?  just seems too paradoxical (genes from single gene)

Reply:
Well, the fusions are indeed queries that involve multiple genes. From the point of OQL fusions are still a query that concern 1 gene, but asks events where this gene may be fuses to a specific other genes. I do not object to renaming it though. Lets discuss.

Reply:
i don't want delay but this name would so puzzle me.  what about getInvolvedGenesFromSingleGeneQuery?